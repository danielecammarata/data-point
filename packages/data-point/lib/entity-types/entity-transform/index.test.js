/* eslint-env jest */

const core = require("../../core");
const factory = require("../entity-transform/factory");
const transformIndex = require("./index");

describe("API", () => {
  it("should match API", () => {
    // eslint-disable-next-line global-require
    expect(transformIndex).toMatchSnapshot();
  });
});

describe("integration", () => {
  it("should create and resolve an entity-transform", async () => {
    const dataPoint = core.create();
    const fooReducer = factory.create("$greeting");
    const input = {
      greeting: "HelloWorld"
    };
    const value = await dataPoint.resolve(fooReducer, input);
    expect(value).toEqual("HelloWorld");
  });
});
