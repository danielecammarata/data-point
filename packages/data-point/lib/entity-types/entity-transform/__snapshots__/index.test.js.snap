// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API should match API 1`] = `[Function]`;
